pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -241,9 +265,25 @@ def save_checkpoint(self, trainer, pl_module):
         # Mode 2: save the last checkpoint
         self._save_last_checkpoint(trainer, pl_module, monitor_candidates)
 
+    def _should_skip_saving_checkpoint(self, trainer) -> bool:
+        return (
+            trainer.fast_dev_run  # disable checkpointing with fast_dev_run
+            or trainer.running_sanity_check  # don't save anything during sanity check
+            or self.save_top_k == 0  # no models are saved
+            or self._last_global_step_saved == trainer.global_step  # already saved at the last step
+        )
+
     def __validate_init_configuration(self):
         if self.save_top_k is not None and self.save_top_k < -1:
             raise MisconfigurationException(f'Invalid value for save_top_k={self.save_top_k}. Must be None or >= -1')
+        if self.every_n_epochs <= -1:
```

Comment:
What `every_n_epochs=0` and `every_n_batches=0` means ? Might be better to start from `<=0`

Reply:
this would be a misconfiguration error as we wouldn't save any checkpoints. as @carmocca said:
- `every_n_*` = 0: disabled
- `every_n_*` > 0: enabled
- `every_n_*` < 0: misconfiguration